src/main/kotlin/org/scoverage/ScoverageAggregate.kt

```diff
@@ -0,0 +1,74 @@
+package org.scoverage
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.FileCollection
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.Nested
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.TaskAction
+import scoverage.report.CoverageAggregator
+
+import org.gradle.api.tasks.PathSensitivity.RELATIVE
+import java.io.File
+
+abstract class ScoverageAggregate: DefaultTask() {
+
+    @Nested
+    var runner: ScoverageRunner? = null
+
+    @InputFiles
+    @PathSensitive(RELATIVE)
+    val sources: ConfigurableFileCollection = project.objects.fileCollection()
+
+    @OutputDirectory
+    val reportDir: Property<File> = project.objects.property(File::class.java)
```

Comment:
```suggestion
    val reportDir: DirectoryProperty = project.objects.directoryProperty()
```
`DirectoryProperty` or `RegularFileProperty`, depending on the usage, are more idiomatic.